perception/autoware_tensorrt_bevdet/src/bevpool_plugin.cu

```diff
@@ -0,0 +1,364 @@
+  // input[2] == ranks_depth      m
+  // input[3] == ranks_feat       m
+  // input[4] == ranks_bev        m
+  // input[5] == interval_starts  k
+  // input[6] == interval_lengths k
+
+  int channel = inputDesc[1].dims.d[1];
+  int n_intervals = inputDesc[5].dims.d[0];
+  int map_size = m_.bev_h * m_.bev_w;
+  int img_size = inputDesc[0].dims.d[2] * inputDesc[0].dims.d[3];
+
+  dim3 grid(GET_BLOCKS(n_intervals * channel));
+  dim3 block(NUM_THREADS);
+
+  switch (int(outputDesc[0].type)) {
+    case int(DataType::kFLOAT):
+      if (inputDesc[0].type == DataType::kFLOAT) {
+        // fp32 fp32
+        bev_pool_v2_kernel<float, float><<<grid, block, 0, stream>>>(
+          channel, n_intervals, map_size, img_size, reinterpret_cast<const float *>(inputs[0]),
+          reinterpret_cast<const float *>(inputs[1]), reinterpret_cast<const int *>(inputs[2]),
+          reinterpret_cast<const int *>(inputs[3]), reinterpret_cast<const int *>(inputs[4]),
+          reinterpret_cast<const int *>(inputs[5]), reinterpret_cast<const int *>(inputs[6]),
+          reinterpret_cast<float *>(outputs[0]));
+      } else {
+        // fp16 fp32
+        bev_pool_v2_kernel<__half, float><<<grid, block, 0, stream>>>(
+          channel, n_intervals, map_size, img_size, reinterpret_cast<const __half *>(inputs[0]),
+          reinterpret_cast<const __half *>(inputs[1]), reinterpret_cast<const int *>(inputs[2]),
+          reinterpret_cast<const int *>(inputs[3]), reinterpret_cast<const int *>(inputs[4]),
+          reinterpret_cast<const int *>(inputs[5]), reinterpret_cast<const int *>(inputs[6]),
+          reinterpret_cast<float *>(outputs[0]));
+      }
+      break;
+    case int(DataType::kHALF):
+      if (inputDesc[0].type == DataType::kFLOAT) {
+        // fp32 fp16
+        bev_pool_v2_kernel<float, __half><<<grid, block, 0, stream>>>(
+          channel, n_intervals, map_size, img_size, reinterpret_cast<const float *>(inputs[0]),
+          reinterpret_cast<const float *>(inputs[1]), reinterpret_cast<const int *>(inputs[2]),
+          reinterpret_cast<const int *>(inputs[3]), reinterpret_cast<const int *>(inputs[4]),
+          reinterpret_cast<const int *>(inputs[5]), reinterpret_cast<const int *>(inputs[6]),
+          reinterpret_cast<__half *>(outputs[0]));
+      } else {
+        // fp16 fp16
+        bev_pool_v2_kernel<__half, __half><<<grid, block, 0, stream>>>(
+          channel, n_intervals, map_size, img_size, reinterpret_cast<const __half *>(inputs[0]),
+          reinterpret_cast<const __half *>(inputs[1]), reinterpret_cast<const int *>(inputs[2]),
+          reinterpret_cast<const int *>(inputs[3]), reinterpret_cast<const int *>(inputs[4]),
+          reinterpret_cast<const int *>(inputs[5]), reinterpret_cast<const int *>(inputs[6]),
+          reinterpret_cast<__half *>(outputs[0]));
+      }
+      break;
+    default:  // should NOT be here
+      RCLCPP_ERROR(rclcpp::get_logger("BEVPoolPlugin"), "\tUnsupport datatype!");
```

Comment:
Instead of using `RCLCPP_ERROR`, which is typically used for ROS logging, consider using a standard C++ error handling mechanism such as throwing an exception or returning an error code.  This would provide a more consistent approach across the codebase and improve integration with other non-ROS components.